unstructured_platform_plugins/schema/json_schema.py

```diff
@@ -29,7 +30,9 @@ def is_generic_alias(val: Any) -> bool:
 
 
 def is_typed_dict(val: Any) -> bool:
-    return inspect.isclass(val) and dict in inspect.getmro(val) and dict != inspect.getmro(val)[0]
+    return (
+        inspect.isclass(val) and dict in inspect.getmro(val) and inspect.getmro(val)[0] is not dict
+    )
```

Comment:
No, this still return True if the dict is an instance of typings `TypedDict`

Reply:
Hmm I meant if the passed val is plain old untyped dict. Anyway, it would be good to add comment about the logic, and supported cases (dict, typed dict, TypedDict) and expected results, as IMHO the implementation is not trivial to figure out how it's working.